src/reconcile/rec_visibility.c

```diff
@@ -488,56 +464,19 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
+                    upd = upd->next;
 
                 WT_ASSERT(session, upd->next == NULL || upd->next->txnid != WT_TXN_ABORTED);
                 upd_select->upd = upd = upd->next;
-
-                /*
-                 * If there is no on-disk update and any valid update from a different transaction
-                 * is not found in the update list, write the same transaction update itself to disk
-                 * to avoid blocking the eviction.
-                 */
-                if (vpack == NULL && upd == NULL)
-                    upd_select->upd = upd = same_txn_valid_upd;
-                else if (upd != NULL && upd->type == WT_UPDATE_TOMBSTONE) {
-                    /*
-                     * The selected update from a different transaction is also a tombstone, use the
-                     * update from the same transaction as the selected update.
-                     */
-                    WT_ASSERT(session,
-                      same_txn_valid_upd != NULL &&
-                        same_txn_valid_upd->type != WT_UPDATE_TOMBSTONE);
-                    upd_select->upd = upd = same_txn_valid_upd;
-                } else if (same_txn_valid_upd != NULL && vpack != NULL &&
-                  WT_TIME_WINDOW_HAS_STOP(&vpack->tw)) {
-                    /*
-                     * The on-disk version has a valid stop timestamp, use the update from the same
-                     * transaction as the selected update.
-                     */
-                    WT_ASSERT(session, same_txn_valid_upd->type != WT_UPDATE_TOMBSTONE);
-                    upd_select->upd = upd = same_txn_valid_upd;
-
-                } else if (same_txn_valid_upd != NULL && vpack != NULL && vpack->tw.prepare) {
-                    /*
-                     * The on-disk version is from an aborted prepare transaction. Therefore, use
-                     * the update from the same transaction as the selected update. We are sure that
-                     * the on-disk prepared update has been aborted because otherwise we would have
-                     * chosen it as an update this tombstone can be applied to.
-                     */
-                    WT_ASSERT(session, same_txn_valid_upd->type != WT_UPDATE_TOMBSTONE);
-                    upd_select->upd = upd = same_txn_valid_upd;
-                }
             }
         }
+
         if (upd != NULL)
             /* The beginning of the validity window is the selected update's time point. */
             WT_TIME_WINDOW_SET_START(select_tw, upd);
         else if (select_tw->stop_ts != WT_TS_NONE || select_tw->stop_txn != WT_TXN_NONE) {
-            /*
-             * We only have a tombstone on the update list or all the updates are from the same
-             * transaction.
-             */
+            /* We only have a tombstone on the update list. */
             WT_ASSERT(session, tombstone != NULL);
 
             /* We must have an ondisk value and it can't be a prepared update. */
```

Comment:
I believe there is another similar looking comment further down. Can you fix it as well?

                /*
                 * Tombstone is the only non-aborted update on the update chain or all the updates
                 * are from the same transaction.
                 */

Reply:
Thanks for spotting it. Fixed and also updated the test_prepare08 to write prepared updates to disk with cursor reset.